### ISSUE TITLE
Mockito.spy(Activity).getBaseContext() returns null on Robolectric 4.4

### ISSUE THREAD

**Issue by ganadist** [robolectric/robolectric#5916]
### Description
Since robolectric 4.4, Mockito.spy(Activity).getBaseContext() returns null

### Steps to Reproduce
```
$ git clone -b mockito_spy_robolectric_4_4 https://github.com/ganadist/VersionCodeDemo demo
$ cd demo

// switch to robolectric 4.3.1
$ git checkout HEAD~1
$ ./gradlew :app:testDevelopDebugUnitTest
BUILD SUCCESSFUL in 14s
28 actionable tasks: 28 executed

// switch to robolectric 4.4
$ git checkout mockito_spy_robolectric_4_4
$ ./gradlew :app:testDevelopDebugUnitTest
> Task :app:testDevelopDebugUnitTest FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':app:testDevelopDebugUnitTest'.
```
Failed unittest result : https://scans.gradle.com/s/wbmmbnzatcyai/tests/:app:testDevelopDebugUnitTest/com.example.myapplication.UnitTest1/test1#1

### Robolectric & Android Version
- Compile Sdk : 29
- Target Sdk : 28
- Robolectric : 4.4 

### Link to a public git repo demonstrating the problem:
https://github.com/ganadist/VersionCodeDemo/blob/mockito_spy_robolectric_4_4/app/src/test/java/com/example/myapplication/UnitTest1.java#L27-L38

**Comment by hoisie** [comment]
Interestingly this passes with Java 11 but fails with Java 8. To me this feels more like a Mockito issue. Would you be able to file an issue here: https://github.com/mockito/mockito/issues? We can collaborate with the Mockito folks to figure out what's happening.

**Comment by ganadist** [comment]
I found this can be failed on JDK11.

I tried to test with various JDK 11, and I found they produced flaky test result

* OpenJDK11 (11.0.1+13) : Flaky
https://scans.gradle.com/s/mbc6chgqgunaq/tests/:app:testDevelopDebugUnitTest/com.example.myapplication.UnitTest1/test1#1
* OpenJDK11 (11.0.2+9) : Flaky
https://scans.gradle.com/s/fuhwugg6447ro/tests/:app:testDevelopDebugUnitTest/com.example.myapplication.UnitTest1/test1#1
* OpenJDK11 (11.0.8+10) : Flaky
https://scans.gradle.com/s/k2x45rgge4j2s/tests/:app:testDevelopDebugUnitTest/com.example.myapplication.UnitTest1/test1#1

* JDK11 from GraalVM 20.2 (11.0.8+10-jvmci-20.2-b03) : Flaky, but it seems to be failed often than vanilla openjdk.
https://scans.gradle.com/s/ovl5yf6popdko/tests/:app:testDevelopDebugUnitTest/com.example.myapplication.UnitTest1/test1#1
https://scans.gradle.com/s/wkx32beocfucm/tests/:app:testDevelopDebugUnitTest/com.example.myapplication.UnitTest1/test1#1

build script modification:
```
$ git diff .
diff --git a/app/build.gradle b/app/build.gradle
index d517378..c80eef2 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -1,8 +1,9 @@
-apply plugin: 'com.android.application'
-
-apply plugin: 'kotlin-android'
-
-apply plugin: 'kotlin-android-extensions'
+plugins {
+  id 'com.android.application'
+  id 'kotlin-android'
+  id 'kotlin-android-extensions'
+  id 'org.gradle.test-retry' version '1.1.9'
+}
 
 android {
     compileSdkVersion 29
@@ -44,6 +45,13 @@ android {
         unitTests {
             includeAndroidResources = true
         }
+       unitTests.all {
+            retry {
+                maxRetries = 3
+                maxFailures = 20
+                failOnPassedAfterRetry = true
+            }
+       }
     }
 }
 
```

Each test performed with following command
```
$ ./gradlew clean
$ env JAVA_HOME=/usr/lib/jvm/xxx ./gradlew :app:testDevelopDebugUnitTest --continue --no-build-cache --scan
```

**Comment by hoisie** [comment]
Thanks for all the work doing the repro and creating an issue on Mockito. Since it is 100% reproducible on Java 8, we should be able to track down the root cause.

**Comment by hoisie** [comment]
Added a simple test case for this issue here: https://github.com/robolectric/robolectric/pull/5925.

**Comment by hoisie** [comment]
This will be fixed in https://github.com/mockito/mockito/pull/2046

**Comment by hoisie** [comment]
Marking as fixed, you should be update to mockito 2.5.11 now.
